common/scala/src/main/scala/org/apache/openwhisk/connector/kafka/KafkaMessagingProvider.scala

```diff
@@ -68,19 +68,25 @@ object KafkaMessagingProvider extends MessagingProvider {
         val nt = new NewTopic(topic, partitions, kafkaConfig.replicationFactor).configs(topicConfig.asJava)
 
         def createTopic(retries: Int = 5): Try[Unit] = {
-          Try(client.createTopics(List(nt).asJava).values().get(topic).get())
-            .map(_ => logging.info(this, s"created topic $topic"))
-            .recoverWith {
-              case CausedBy(_: TopicExistsException) =>
-                Success(logging.info(this, s"topic $topic already existed"))
-              case CausedBy(t: RetriableException) if retries > 0 =>
-                logging.warn(this, s"topic $topic could not be created because of $t, retries left: $retries")
-                Thread.sleep(1.second.toMillis)
-                createTopic(retries - 1)
-              case t =>
-                logging.error(this, s"ensureTopic for $topic failed due to $t")
-                Failure(t)
-            }
+          Try(client.listTopics().names().get())
+            .map(topics =>
+              if (topics.contains(topic)) {
+                Success(logging.info(this, s"$topic already exists and the user can see it, skipping creation."))
+              } else {
+                Try(client.createTopics(List(nt).asJava).values().get(topic).get())
+                  .map(_ => logging.info(this, s"created topic $topic"))
+                  .recoverWith {
+                    case CausedBy(_: TopicExistsException) =>
```

Comment:
I'm wondering if we could extract the exception that is thrown when the `controllers / invokers` are configured with lower privileged users and display a more personalized message for developer, that would help them realize that the topics have not been created prior to OW deployment but at the same time they can't be created because of permission issues.

Reply:
I didn't want to use exceptions for flow control.
Right now it will throw a message that says:
```
[2020-01-23T13:03:59.094Z] [INFO] [#tid_sid_unknown] [KafkaMessagingProvider] completedstandalone already exists and the user can see it, skipping creation.
[2020-01-23T13:04:00.383Z] [ERROR] [#tid_sid_unknown] [KafkaMessagingProvider] ensureTopic for health failed due to java.util.concurrent.ExecutionException: org.apache.kafka.common.errors.TopicAuthorizationException: Topic authorization failed.
```

Reply:
👍 yeah, it seems that the message is pretty obvious